src/app/store/test-records/effects/test-records.effects.ts

```diff
@@ -118,6 +155,41 @@ export class TestResultsEffects {
     )
   );
 
+  /**
+   * Call PUT Test Results API to cancel test result
+   */
+  cancelTestResult$ = createEffect(() =>
```

Comment:
it seems strange to me that we need a whole new effect for what is effectively the exact same as an update. Would it now work to subscribe to the value changes of the form, update the editingTestResult and then just dispatch the usual test result update?

Reply:
An example of your suggestion?

But in hindsight, that would go against the dummy component approach. Components, ideally, shouldn't need to how editingTestResult works and how it links to our usual update flow. They should just need to dispatch an action and don't worry about anything else. I would rather have 2 actions that are similar than put any more responsibility on the components.

Reply:
Suggestion applied.